instrumentation/messaging/src/main/java/brave/messaging/MessagingResponse.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2013-2020 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package brave.messaging;
+
+import brave.Response;
+import brave.internal.Nullable;
+
+/**
+ * Abstract response type used for parsing and sampling of messaging clients and servers.
+ *
+ * @see ProducerResponse
+ * @see ConsumerResponse
+ * @since 5.13
+ */
+public abstract class MessagingResponse extends Response {
+  /**
+   * The request that initiated this messaging response or {@code null} if unknown.
```

Comment:
Curious when this could be set - if the MQ is run in-process?

Reply:
probably not the best sentence. So, for example the messageId field is only readable during the response processing of AWS SQS.

Different than RPC I think here it isn't important that the request is what initiated the response, rather it is simply the start of the operation, available here so you can late process data like message Id

Reply:
Ah I think it could be `Information about the request that...`